File: 2022/1/solution.py
import os
from typing import List

def parse(puzzle_input: List[str]) -> List[int]:
  elves: List[int] = []
  curr = 0
  for line in puzzle_input:
    if line:
      curr += int(line)
    else:
      elves.append(curr)
      curr = 0
  elves.append(curr)
  return elves

if __name__ == '__main__':
  with open(os.path.join(os.path.dirname(os.path.abspath(__file__)), 'input.txt'),'r') as file:
    puzzle_input = [line.strip() for line in file]
  elves = parse(puzzle_input)
  elves.sort()
  print(elves[-1])
  print(sum(elves[-3:]))

File: 2022/1/solution.js
const fs = require("fs");
const path = require("path");

const input = fs
  .readFileSync(path.join(__dirname, "input.txt"), "utf-8")
  .split("\n")
  .map((a) => (a.length ? Number(a) : ""));

input.push("");
// let part1 = 0;
// let curr = 0;
// for (const line of input) {
//   if (line === "") {
//     part1 = Math.max(part1, curr);
//     curr = 0;
//   } else {
//     curr += line;
//   }
// }
// console.log(part1);

const elves = [];
let curr = 0;
for (const line of input) {
  if (line === "") {
    elves.push(curr);
    curr = 0;
  } else {
    curr += line;
  }
}
elves.sort((a, b) => b - a);
console.log(elves[0]);
console.log(elves[0] + elves[1] + elves[2]);
